src/tool_getparam.c

```diff
@@ -1630,6 +1678,16 @@ ParameterError getparameter(const char *flag, /* f or -long-flag */
     case C_TCP_NODELAY: /* --tcp-nodelay */
       config->tcp_nodelay = toggle;
       break;
+    case C_IP_TOS: { /* --ip-tos */
+      const struct TOSEntry *entry = bsearch(
+        &nextarg, tos_entries, sizeof(tos_entries)/sizeof(*tos_entries),
+        sizeof(*tos_entries), find_tos);
```

Comment:
Just curious, is declaring a variable in a `case` block valid C89? I was under the impression that it isn't, and curl also used explicit blocks in some other code in similar cases.

Reply:
It's in a new block.

Reply:
Declaring *and initializing* a variable in a case that has subsequent cases is considered undefined behavior (in all C/C++ variants), and GCC warns about it.

Reply:
It'd make sense, but the pattern often (?) seen is this:
```c
case <n>:
{
   int var = 1;
}
```
`case` being a block is not 100% self-evident: https://en.wikipedia.org/wiki/Duff%27s_device

Hence the question.

Reply:
But there's a brace there. The same style is used all over this file. Examples:
```c
case C_PARALLEL_MAX: {  /* --parallel-max */
case C_LOCAL_PORT: { /* --local-port */
case C_REFERER: { /* --referer */
```

Reply:
I apologize for the noise, indeed there is, I didn't notice!